Scripts/script-IsEmailAddressInternal.yml

```diff
@@ -6,43 +6,50 @@ script: |-
   import re
   from distutils.util import strtobool
 
-  email = demisto.args()['email']
+  emails = [x.lower() for x in argToList(demisto.get(demisto.args(),'email'))]
   domains = [x.lower() for x in argToList(demisto.get(demisto.args(),'domain'))]
   include_subdomains = strtobool(demisto.getArg('include_subdomains') or 'no')
 
-  parts = email.split('@',1)
+  results = []
+  ec = {}
+
+  for email in emails:
+      parts = email.split('@',1)
+      network_type = "Unknown"
+      if len(parts) > 1:
+          if parts[1].lower() in domains or include_subdomains and re.match("^(.*\\.)?({})".format('|'.join([re.escape(d) for d in domains])), parts[1].lower()):
+              network_type = "Internal"
+          else:
+              network_type = "External"
+
+          email_dict = {
+              "Address" : email,
+              "Domain" : parts[1],
+              "Username": parts[0],
+              "NetworkType" : network_type
+          }
+
+          ec['Account.Email(val.Address && val.Address == ' + email_dict["Address"] + ')'] = email_dict
```

Comment:
save the dict in a list and at the end use the original dt (comparing to `obj.Address` instead of dynamic condition)